service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java

```diff
@@ -172,6 +171,22 @@ public PublishedMetadbSample getPublishedMetadbSample(UUID metadbSampleId) throw
         return samples;
     }
 
+    @Override
+    public List<MetadbSample> getMetadbSampleListByCmoPatientId(String cmoPatientId) throws Exception {
+        List<SampleMetadata> sampleMetadataList = sampleRepository
+                .findAllSampleMetadataListByCmoPatientId(cmoPatientId);
+        List<MetadbSample> samples = new ArrayList<>();
+        for (SampleMetadata sample: sampleMetadataList) {
+            // TODO: update method to fill in sample details in a more inclusive
+            // way and not just request and igo id since that is very specific
+            // to research samples only
+            MetadbSample metadbSample = sampleRepository.findResearchSampleByRequestAndIgoId(
```

Comment:
Should we add the clinical sample update handling, since it would use the same methods as research samples now that they are both in the universal schema form?

Reply:
I don't think we should before introducing the other clinical dev work